sherpa/ui/utils.py

```diff
@@ -332,19 +333,85 @@ def reduce_ufunc(func):
 
 
 class ModelWrapper(NoNewAttributesAfterInit):
+    """Wrap up a model class so we can create instances easily.
+
+    The created instances can be used in a "model language" to create
+    complex expressions. If mdl1 and mdl2 have been created by
+    ModelWrapper then a user can say mdl1.n1 + mdl2.n2 to create model
+    instances named n1 and n2 and then return the expression which
+    represents their sum. Note that the model component is stored in
+    the session object.
+
+    """
+
+    def __init__(self, session, modeltype, args=(), kwargs=None):
+        # This is an internal class so do not bother with
+        # sherpa.utils.err exceptions.
+        #
+        if not isinstance(session, Session):
+            raise ValueError("session is not a Session instance")
+
+        if not _is_subclass(modeltype, Model):
+            raise ValueError("modeltype is not a Model instance")
 
-    def __init__(self, session, modeltype, args=(), kwargs={}):
         self._session = session
         self.modeltype = modeltype
         self.args = args
-        self.kwargs = kwargs
+        self.kwargs = kwargs if kwargs else {}
+
+        # Edit the docstring of the new object to hide the
+        # ModelWrapper text and replace it with a reference to the
+        # wrapped class.
+        #
+        mname = modeltype.__name__.lower()
+        prefix = "an" if mname[0] in "aeiou" else "a"
```

Comment:
Nice attention to detail! I'm not asking for code to detect a silent H or a Y at the beginning of the model name...